app/controllers/people/person/personal-information/edit.js

```diff
@@ -1,23 +1,49 @@
 import Controller from '@ember/controller';
 import { dropTask } from 'ember-concurrency';
 import { inject as service } from '@ember/service';
+import { combineFullAddress } from 'frontend-contact-hub/models/address';
 
 export default class PeoplePersonPersonalInformationEditController extends Controller {
   @service router;
 
   @dropTask
   *save(event) {
     event.preventDefault();
+    let { person, contacts } = this.model;
+    yield person.validate();
+    let validContacts = true;
 
-    yield this.model.person.validate();
+    for (let contact of contacts) {
+      let { primaryContact, secondaryContact, address } = contact;
+      yield primaryContact.validate();
+      yield secondaryContact.validate();
+      yield address.validate();
+      if (
+        primaryContact.isValid &&
+        secondaryContact.isValid &&
+        address.isValid
+      ) {
+        if (address.isDirty) {
+          address.fullAddress = combineFullAddress(address);
+          yield address.save();
+        }
+        primaryContact.contactAddress = address;
+        if (primaryContact.isDirty) {
+          yield primaryContact.save();
+        }
+        if (secondaryContact.isDirty) {
+          yield secondaryContact.save();
+        }
+      } else {
+        validContacts = false;
+        break;
+      }
+    }
 
-    if (this.model.person.isValid) {
-      yield this.model.person.save();
+    if (validContacts && person.isValid) {
```

Comment:
I would change the flow a bit. It's still possible that some records are being saved while not everything is valid.

I would first loop through all the contacts to see if there are any invalid records. If there aren't any and the person is also valid the saving logic can run.